.ci/Makefile

```diff
@@ -18,6 +21,14 @@ clean:
 env:
 	env
 
+# Create Docker image
+build-docker: validate-branch-name
+	docker build -t $(DOCKER_REGISTRY)/$(DOCKER_IMAGE_NAME):$(BRANCH_NAME) \
```

Comment:
Does the BRANCH_NAME include the `v` prefix, as in `v3.1.2`? If so, we probably want to remove it.

Reply:
Probably beyond the scope of this but we might also want to publish a `latest` tag and a tag for the latest major version, such as `3.x`. Not that `latest` doesn't necessarily refer to the most recent release, but the one that has the highest semver version number. For example `3.1.0 > 3.0.1`, even if `3.0.1` has been released after `3.1.0`.

Reply:
In current usage the `BRANCH_NAME` does *not* include the `v` prefix.

Reply:
+1 to "latest". We won't have much need for a "3.x" tag until/if we do a "4.x" major version of the APM agent.

Reply:
> Does the BRANCH_NAME include the v prefix, as in v3.1.2? If so, we probably want to remove it.

Yes, it does. Will try to remove it.

> Probably beyond the scope of this but we might also want to publish a latest tag ...

I added pushing of the `latest` tag already in `dev-utils/push-docker.sh`, but without the advanced logic, yet. 
Do you know how we can query the tags on a docker repo (e.g. from something like this https://container-library.elastic.co/r/observability/apm-agent-java ) to implement this advanced logic?

Reply:
| In current usage the BRANCH_NAME does not include the v prefix.

@trentm With my local testing it did. But I guess it behaves different on CI?

Reply:
If it helps with the logic, I don't think we've *ever* done the equivalent of a "3.0.1" release after having already released "3.1.0" or later. I.e. I don't think this code needs to worry about whether the current release *is* the "latest". Perhaps just have a "IS_LATEST" var hardcoded to true. If/when we have a "3.x" maintenance branch and "main" is for version 4.x of the agent, then we can set `IS_LATEST=false` for the 3.x branch (or just delete the block of code that adds the "latest" tag).

Reply:
> Do you know how we can query the tags on a docker repo (e.g. from something like this https://container-library.elastic.co/r/observability/apm-agent-java ) to implement this advanced logic?

This script returns the latest version number:
```
git tag --list "v*" | cut -c2- | sort --version-sort | tail -n 1
```

You could just check whether the current version is the same.

Reply:
Another alternative would be `git tag --list --sort=version:refname "v*" | cut -c2- | tail -n1`

If the tags may contain suffix that indicates pre-releases, such as `1.2.3-pre1`, or `1.2.3.RC1` (the latter is used in the Java agent) we can use this: `git -c 'versionsort.suffix=.RC' tag --list --sort=version:refname "v*" | cut -c2- | tail -n1`